stress-example/image_search/client.py

```diff
@@ -34,40 +38,48 @@ def random_docs(start, end):
         yield doc
 
 
-docs_index = random_docs(0, NUM_DOCS)
-docs_query = random_docs(0, QUERY_NUM_DOCS)
-
-
-def index(port):
-    # TODO get hostname and port programatically
-    host = '0.0.0.0'
-    args = set_client_cli_parser().parse_args(
-        ['--host', host, '--port-expose', str(port)])
-    grpc_client = Client(args)
-    grpc_client.index(docs_index)
+def index(client, docs):
+    client.index(docs)
 
 
-def query(port):
-    # TODO get hostname and port programatically
-    host = '0.0.0.0'
-    args = set_client_cli_parser().parse_args(
-        ['--host', host, '--port-expose', str(port)])
-    grpc_client = Client(args)
-    grpc_client.search(input_fn=docs_query, on_done=validate_img, top_k=TOP_K)
+def query(client, docs):
+    client.search(input_fn=docs, on_done=validate_img, top_k=TOP_K)
 
 
 @click.command()
 @click.option('--task', '-t')
 @click.option('--port', '-p')
-def main(task, port):
-    if task == 'index':
-        index(port)
-    elif task == 'query':
-        query(port)
-    else:
+@click.option('--load', '-l', default=60)  # time (seconds)
+@click.option('--nr', '-n', default=DEFAULT_NUM_DOCS)
+def main(task, port, load, nr):
```

Comment:
do we want to set the parallelization at this level?

Reply:
yes will add it now. Committed this as the refactoring required to do that